drivers/vfio/pci/nvgrace-gpu/egm.c

```diff
@@ -365,6 +363,7 @@ int register_egm_node(struct pci_dev *pdev)
 	region->egmlength = egmlength;
 	region->egmpxm = egmpxm;
 
+	hash_init(region->htbl);
```

Comment:
It was moved because previously it was located in nvgrace_egm_fetch_bad_pages() and in a place where that routine could return (legitimately w/o error) without having performed the initialization. If that were to occur, in other places where the table is referenced (e.g. tear down path, ioctl path, fault patch, etc.) the lack of initialization may lead to unintended effects. It makes sense to initialize it alongside "region" allocation and other initializers.

Reply:
Make sense